### DISCUSSION TITLE
How work {function} in v5

### DISCUSSION THREAD

**Discussion by criocere** [smarty-php/smarty#1070] [Q&A] [answered]
I've tried to migrate from v4 to v5 but I got an error on all my template functions.
Example

```
{function selection label='' column='' value='' options=''}
.....
{/function}
```

I got an error when trying to use it where it was ok in v4

`Smarty Compiler: Syntax error in template "file:myTemplate.tpl"  on line 13 "{selection label= ....}`

I saw nothing in the doc for this kind of functions

**Comment by criocere** [comment]
Ho I saw that I have to replace {myTplFunction} by {call myTplFuncion}

Mybe this can be added in the doc?
https://smarty-php.github.io/smarty/stable/upgrading/

**Reply by wisskid** [reply]
I think this change is unintentional. I'll have a look if we can fix this.

**Reply by wisskid** [reply]
I cannot reproduce your error, @criocere . Can you post a minimal example that triggers the error?

**Comment by criocere** [comment]
I simplify my code and I reproduce the issue with this code

zzzzChild.tpl
```
{include file='inc/zzzz.inc.tpl'}
{selection label='Compte' table='Comptes' columns='RaisonSociale' column='selectionQuery'} 
```

zzzz.php
```
<?php
include('../inc/define.inc.php');
include(WEBROOT.'smarty/libs/Smarty.class.php');

use Smarty\Smarty;
$smarty = new Smarty();

$smarty->compile_check = true;
$smarty->debugging = true;

$smarty->setCaching(Smarty::CACHING_OFF);
$smarty->merge_compiled_includes = true; // perfos

$smarty->setCacheLifetime(-1);
$smarty->setTemplateDir(WEBROOT.'templates');
$smarty->setCompileDir(WEBROOT.'cache/templates_c');
$smarty->setCacheDir(WEBROOT.'cache/cache');
$smarty->setConfigDir(WEBROOT.'cache/configs');
$phpPlugins = ['ceil', 'time', 'strval', 'strtotime', 'ucwords', 'mb_convert_encoding', 'date', 'hash', 'urlencode', 'strtok', 'htmlspecialchars', 'html_entity_decode', 'stripslashes', 'strtolower', 'strtoupper', 'strpos', 'ucfirst', 'md5', 'implode', 'explode', 'addslashes', 'intval', 'strip_tags', 'utf8_encode', 'substr', 'json_encode', 'trim', 'rand'];
foreach($phpPlugins as $el) $smarty->registerPlugin("modifier",$el, $el);

$smarty->display('zzzzChild.tpl');
?>
```

inc/zzzz.inc.php
```
{function selection label='' table='' columns='' column=''}
    <p>Hello World!</p>
{/function}
```

Error is
`Fatal error: Uncaught --> Smarty Compiler: Syntax error in template "file:zzzzChild.tpl" on line 2 "{selection label='Compte' table='Comptes' columns='RaisonSociale' column='selectionQuery'}" unknown tag 'selection' <-- thrown in /home/adn4you1/domains/acdf.adn4you.com/private_html/gdc/templates/zzzzChild.tpl on line 2`

I think I found the problem, if I replace the include by the code it works with this

zzzzChild.tpl
```
{function selection label='' table='' columns='' column=''}
    <p>Hello World!</p>
{/function}
{selection label='Compte' table='Comptes' columns='RaisonSociale' column='selectionQuery'}  
```

**Reply by wisskid** [reply, marked as answer]
The documentation states that if you define a template function in a separate template, you need to use the `{call name="selection"}` syntax to invoke it. This does indicate a change in behavior regarding inlining of included templates. I know this has changed in v5.

**Reply by criocere** [reply]
As it works sinces years, I never see that the call function has to be used in this case.
Thanks for your answer
